src/icmp.rs

```diff
@@ -0,0 +1,353 @@
+                0x0B => Self::DestinationUnreachable(Unreachable::NetworkUnreachableForTos),
+                0x0C => Self::DestinationUnreachable(Unreachable::HostUnreachableForTos),
+                0x0D => Self::DestinationUnreachable(Unreachable::CommunicationAdministrativelyProhibited),
+                0x0E => Self::DestinationUnreachable(Unreachable::HostPrecedenceViolation),
+                0x0F => Self::DestinationUnreachable(Unreachable::PrecedentCutoffInEffect),
+                _ => Self::Other(raw),
+            },
+            0x04 => match c {
+                0x00 => Self::SourceQuench,
+                _ => Self::Other(raw),
+            },
+            0x05 => match c {
+                0x00 => Self::Redirect(Redirect::Network),
+                0x01 => Self::Redirect(Redirect::Host),
+                0x02 => Self::Redirect(Redirect::TosAndNetwork),
+                0x03 => Self::Redirect(Redirect::TosAndHost),
+                _ => Self::Other(raw),
+            },
+            0x07 => Self::Reserved,
+            0x08 => Self::EchoRequest,
+            0x09 => Self::RouterAdvertisment,
+            0x0A => Self::RouterSolicication,
+            0x0B => match c {
+                0x00 => Self::TimeExceeded(TimeExceeded::TTL),
+                0x01 => Self::TimeExceeded(TimeExceeded::FragmentReassembly),
+                _ => Self::Other(raw),
+            },
+            0x0C => match c {
+                0x00 => Self::ParameterProblem(ParameterProblem::Pointer),
+                0x01 => Self::ParameterProblem(ParameterProblem::MissingRequiredOption),
+                0x02 => Self::ParameterProblem(ParameterProblem::BadLength),
+                _ => Self::Other(raw),
+            },
+            0x0D => Self::Timestamp,
+            0x0E => Self::TimestampReply,
+            0x2A => Self::ExtendedEchoRequest,
+            0x2B => match c {
+                0x00 => Self::ExtendedEchoReply(ExtendedEchoReply::NoError),
+                0x01 => Self::ExtendedEchoReply(ExtendedEchoReply::MalformedQuery),
+                0x02 => Self::ExtendedEchoReply(ExtendedEchoReply::NoSuchInterface),
+                0x03 => Self::ExtendedEchoReply(ExtendedEchoReply::NoSuchTableEntry),
+                0x04 => Self::ExtendedEchoReply(ExtendedEchoReply::MupltipleInterfacesStatisfyQuery),
+                _ => Self::Other(raw),
+            }
+            _ => Self::Other(raw),
+        }
+    }
+}
+
+fn parse_icmp_code(input: &[u8]) -> IResult<&[u8], IcmpCode> {
+    let (input, code) = number::streaming::be_u16(input)?;
+
+    Ok((input, code.into()))
+}
+
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+pub enum IcmpData {
+    Unreachable { nexthop_mtu: u16, header: IPv4Header, packet: [u8; 8] },
+    Redirect {gateway: Ipv4Addr, header: IPv4Header, packet: [u8; 8] },
+    TimeExceeded {header: IPv4Header, packet: [u8; 8] },
```

Comment:
missing space

Reply:
as a matter of fact I'd like all the code to go through `rustfmt` before submission

Reply:
generally that the owner of a crate that do that to avoid trust issue

Reply:
I did run it,
the changes along with other fixes are in https://github.com/bestouff/pktparse-rs/pull/12/commits/63056f68094a20de49f9d9b44600cf36b0401e86

Reply:
ah this file only, nevermind, I thought you talk about the project.

Reply:
Running it for all the project is probably out of the scope of this PR.
As you mentioned, it's probably better when the owner does it (before the release for example)

Reply:
Well, I'd say that it's better to move to non v6 first.
Let's wait for https://github.com/bestouff/pktparse-rs/pull/11

Reply:
Sure right, I'll try to remember to do that later